### PR TITLE
autocomplete dropdown- initial version

### PR DESCRIPTION
**Pull request by Erin007**
This generic form component is the first step in building the school drop down menu for the census form. The component takes in an array of options, and as the the user types, displays options that match in a dropdown menu. 

autocomplete-dropdown

### CODE REVIEW

**Comment by Erin007** [comment]
@joannepolsky Not sure yet what format the data returned from your API is, but this takes in an array of objects and iterates through them.  We'll have to work out fetching subsets since the school list is really long.  Also, this matches for the letter combos anywhere in the string (e.g. Caley BRock is a result for "br") so I might have to adjust if that isn't the behavior we want. But, it's a start 😀

File: apps/src/templates/AutocompleteDropdown.story.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    { value: 'MS', label: 'Mehal Shah' },
+    { value: 'EH', label: 'Elijah Hamovitz' },
+    { value: 'JS', label: 'Jeremy Stone' },
+    { value: 'EJ', label: 'Eric Jordan' },
+    { value: 'WJ', label: 'Will Jordan' },
+    { value: 'BB', label: 'Brad Buchanan' },
+    { value: 'AO', label: 'Andrew Oberhardt' },
+    { value: 'DB', label: 'David Bailey' },
+    { value: 'RK', label: 'Ram Kandasamy' },
+  ];
```
Comment (ashercodeorg) [inline apps/src/templates/AutocompleteDropdown.story.js:22]: Prefer not to use PII in our codebase. :)

File: apps/src/templates/AutocompleteDropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+const AutocompleteDropdown = React.createClass({
+  propTypes: {
+    options: PropTypes.array.required,
+  },
+
+  getInitialState() {
+    return {
+    disabled: false,
+    searchable: true,
```
Comment (Bjvanminnen) [inline apps/src/templates/AutocompleteDropdown.jsx:31]: nit: need one more level of indentation on above two lines

File: apps/src/templates/AutocompleteDropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import Select from 'react-select';
+
+const AutocompleteDropdown = React.createClass({
```
Comment (Bjvanminnen) [inline apps/src/templates/AutocompleteDropdown.jsx]: I think we're favoring `extends Component` over `creatClass` on new components these days

**Comment by davidsbailey** [comment]
What you have here seems reasonable in isolation, however it will need the support of a server API that retrieves a list of schools for a given substring because the size of the schools data is too large to pull down to the client in its entirety (100K schools, 6MB, see [dashboard/config/schools.tsv](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/schools.tsv)). 

I don't think there's anything prohibitively difficult about building such a server API, however we should make sure that this work is being accounted for in the schedule. @joannepolsky is that something you're working on? 

Here is how this looks in our existing school info flow during teacher signup. Note matches against both prefixes and substrings:
screen shot 2017-08-31 at 3 06 56 pm

**Comment by davidsbailey** [comment]
Apologies Erin for not acknowledging some of the observations you've made above -- I realize that you are aware that we'll need to chunk the schools, and that we need to filter based on the substring. My other concern (besides scheduling) is that it might be hard to make progress here without knowing what the server API will look like and how we plan to use it. For example, if the server API does the filtering then we might not need to do that here.

**Pull request closed without merging**